WcaOnRails/spec/models/competition_wcif_spec.rb

```diff
@@ -0,0 +1,177 @@
+  let!(:competition) {
+    FactoryGirl.create(
+      :competition,
+      :with_delegate,
+      id: "TestComp2014",
+      name: "Test Comp 2014",
+      start_date: "2014-02-03",
+      end_date: "2014-02-05",
+      external_website: "http://example.com",
+      showAtAll: true,
+      event_ids: %w(333 444),
+    )
+  }
+  let(:delegate) { competition.delegates.first }
+  let(:sixty_second_2_attempt_cutoff) { Cutoff.new(numberOfAttempts: 2, attemptValue: ONE_MINUTE_IN_CENTISECONDS) }
+  let(:top_16_advance) { AdvanceToNextRoundRequirement.new(type: "ranking", ranking: 16) }
+  let!(:round333_1) { FactoryGirl.create(:round, competition: competition, event_id: "333", number: 1, cutoff: sixty_second_2_attempt_cutoff, advance_to_next_round_requirement: top_16_advance) }
+  let!(:round333_2) { FactoryGirl.create(:round, competition: competition, event_id: "333", number: 2) }
+  let!(:round444_1) { FactoryGirl.create(:round, competition: competition, event_id: "444", number: 1) }
+  before :each do
+    competition.reload
+  end
+
+  context "to_wcif" do
+    it "basic check" do
+      expect(competition.to_wcif).to eq(
+        "formatVersion" => "1.0",
+        "id" => "TestComp2014",
+        "name" => "Test Comp 2014",
+        "persons" => [delegate.to_wcif(competition)],
+        "events" => [
+          {
+            "id" => "333",
+            "rounds" => [
+              {
+                "id" => "333-1",
+                "format" => "a",
+                "timeLimit" => {
+                  "centiseconds" => TimeLimit::TEN_MINUTES_IN_CENTISECONDS,
+                  "cumulativeRoundIds" => [],
+                },
+                "cutoff" => {
+                  "numberOfAttempts" => 2,
+                  "attemptValue" => ONE_MINUTE_IN_CENTISECONDS,
+                },
+                "advanceToNextRoundRequirement" => {
+                  "type" => "ranking",
+                  "ranking" => 16,
+                },
+              },
+              {
+                "id" => "333-2",
+                "format" => "a",
+                "timeLimit" => {
+                  "centiseconds" => TimeLimit::TEN_MINUTES_IN_CENTISECONDS,
+                  "cumulativeRoundIds" => [],
+                },
+                "cutoff" => nil,
+                "advanceToNextRoundRequirement" => nil,
+              },
+            ],
+          },
+          {
+            "id" => "444",
+            "rounds" => [
+              {
+                "id" => "444-1",
+                "format" => "a",
+                "timeLimit" => {
+                  "centiseconds" => TimeLimit::TEN_MINUTES_IN_CENTISECONDS,
+                  "cumulativeRoundIds" => [],
+                },
+                "cutoff" => nil,
+                "advanceToNextRoundRequirement" => nil,
+              },
+            ],
+          },
+        ],
+      )
+    end
+  end
+
+  context "load_wcif!" do
+    it "remove event by setting rounds to empty" do
```

Comment:
It's nice if a spec description makes a nice sentence with the rest, so I'd suggest something like `"removes competition event when its rounds are empty"`.

I spent too much time on spec nitpicking, sorry!